eth/eventservices/jobservice.go

```diff
@@ -46,8 +49,17 @@ func (s *JobService) Start(ctx context.Context) error {
 	sub, err := s.eventMonitor.SubscribeNewJob(ctx, "NewJob", logsCh, common.Address{}, func(l types.Log) (bool, error) {
 		_, jid, _, _ := parseNewJobLog(l)
 
-		job, err := s.node.Eth.GetJob(jid)
-		if err != nil {
+		var job *lpTypes.Job
+		getJob := func() error {
+			j, err := s.node.Eth.GetJob(jid)
+			if j.StreamId == "" {
+				glog.Errorf("Got empty job for id:%v. Should try again.", jid.Int64())
+				return errors.New("ErrGetJob")
+			}
+			job = j
+			return err
+		}
+		if err := backoff.Retry(getJob, backoff.NewConstantBackOff(time.Second*2)); err != nil {
 			glog.Errorf("Error getting job info: %v", err)
 			return false, err
```

Comment:
One additional thing I thought of - in that edge case described in the above comment, the transcoder would retry a number of times and then give up and actually stop watching for new job events since it returns `false, err`

Reply:
I think the current logic here is to indefinitely retry every 2 seconds.  I'm actually not sure when we will get an error here, but we do end up getting one, something unexpected has happened.